cmd/cluster/aws/create.go

```diff
@@ -77,10 +85,89 @@ func applyPlatformSpecificsValues(ctx context.Context, exampleOptions *apifixtur
 	if err != nil {
 		return err
 	}
-	infraID := opts.InfraID
-
 	// Load or create infrastructure for the cluster
 	var infra *awsinfra.CreateInfraOutput
+	var iamInfo *awsinfra.CreateIAMOutput
+	infraID := opts.InfraID
+
+	if infraID == "" {
+		return ErrMsg("infra-id")
+	}
+	hci := hyperv1.HostedClusterInfrastructure{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      opts.InfraID,
+			Namespace: opts.Namespace,
+		},
+	}
+	if err := client.Get(ctx, crclient.ObjectKeyFromObject(&hci), &hci); err != nil {
+		if apierrors.IsNotFound(err) {
+			opts.Log.Info("Hosted cluster infrastructure not found, creating infrastructure with user input", "namespace", opts.Namespace, "name", opts.InfraID)
+		} else {
+			return err
+		}
+	} else {
+		opts.Log.Info("Found hosted cluster infrastructure", "namespace", opts.Namespace, "name", opts.InfraID)
+		if hci.Spec.SubDomain != opts.Name {
+			return fmt.Errorf(fmt.Sprintf("The HostedClusterInfrastructure.Spec.SubDomain must equal the cluster name\nCluster name: %s, Sub-Domain: %s", opts.Name, hci.Spec.SubDomain))
+		}
+		psStatus, infraStatus, iamStatus := hostedclusterinfrastructure.GetInfraIamConditionStatus(&hci)
+		if !psStatus || !infraStatus || !iamStatus {
+			return fmt.Errorf("the hosted cluster infrastructure resource is not configured as expected\n- provider-secret  : %t\n- infra-as-expected: %t\n- iam-as-expected  : %t",
+				psStatus, infraStatus, iamStatus)
+		}
+		//Set a finalizer on HCI
+		origHCI := hci.DeepCopy()
+		controllerutil.AddFinalizer(&hci, core.DestroyClusterFinalizer)
+		if hci.ObjectMeta.Labels == nil {
+			hci.SetLabels(map[string]string{})
+		}
+		hci.ObjectMeta.Labels[hostedClusterLabel] = opts.Name
+		if err := client.Patch(ctx, &hci, crclient.MergeFrom(origHCI)); err != nil {
+			return err
+		}
+
+		if opts.ServiceCIDR == "" {
+			exampleOptions.ServiceCIDR = "172.31.0.0/16"
+		}
+		if opts.ClusterCIDR == "" {
+			exampleOptions.ClusterCIDR = "10.132.0.0/14"
+		}
+		if opts.ControlPlaneAvailabilityPolicy == "" {
+			exampleOptions.ControlPlaneAvailabilityPolicy = "SingleReplica"
+		}
+		infra = &awsinfra.CreateInfraOutput{
```

Comment:
Can this all be a hciToInfraOuput() func?

Reply:
If using getInfraFromHCI, do you still want to sub-divide?

Reply:
no longer applies to this pr.